pkg/auth/dependency/interaction/intent.go

```diff
@@ -89,3 +90,11 @@ type IntentRemoveAuthenticator struct {
 }
 
 func (*IntentRemoveAuthenticator) Type() IntentType { return IntentTypeRemoveAuthenticator }
+
+type IntentUpdateAuthenticator struct {
+	Authenticator    authenticator.Spec `json:"authenticator"`
+	OldSecret        string             `json:"-"`
+	RequireOldSecret bool               `json:"-"`
```

Comment:
I think this flag should be reversed (e.g. `SkipVerifySecret`), since most of the time we do want verification. Skipping verification is rare and should be done with awareness, not just by omitting the flag (maybe carelessly).

Reply:
Sure, updated!